patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -2438,6 +2438,16 @@
  
  			new Microsoft.Xna.Framework.Color(150, 150, 150, 150);
  			if (mouseX >= num && (float)mouseX <= (float)num + (float)TextureAssets.InventoryBack.Width() * inventoryScale && mouseY >= num2 && (float)mouseY <= (float)num2 + (float)TextureAssets.InventoryBack.Height() * inventoryScale && !PlayerInput.IgnoreMouseInterface) {
+@@ -31238,7 +_,8 @@
+ 
+ 		public static void CraftItem(Recipe r) {
+ 			int stack = mouseItem.stack;
+-			mouseItem = r.createItem.Clone();
++			mouseItem = r.createItem;
++			r.createItem = new Item(mouseItem.type) { stack = mouseItem.stack };
```

Comment:
I still think this should be the other way
`mouseItem = new Item(mouseItem.type) {stack = r.createItem.stack}`
and leave `r.createItem` alone. 

The default behaviour should be for an item which changes colour on craft, to not change colour in the recipe viewer. People can implement this if they want using OnCraft (move the recipe colour into the newItem, and recolour the recipe item) but it should be opt_in_

Reply:
I suppose, I can add it to the ExampleInstanceItem.